core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/queries/CassandraSchemaQueries.java

```diff
@@ -131,70 +132,89 @@ private void executeOnAdminExecutor() {
 
     schemaRowsBuilder = new CassandraSchemaRows.Builder(isCassandraV3, refreshFuture, logPrefix);
 
-    query(selectKeyspacesQuery() + whereClause, schemaRowsBuilder::withKeyspaces);
-    query(selectTypesQuery() + whereClause, schemaRowsBuilder::withTypes);
-    query(selectTablesQuery() + whereClause, schemaRowsBuilder::withTables);
-    query(selectColumnsQuery() + whereClause, schemaRowsBuilder::withColumns);
+    query(selectKeyspacesQuery() + whereClause, schemaRowsBuilder::withKeyspaces, true);
+    query(selectTypesQuery() + whereClause, schemaRowsBuilder::withTypes, true);
+    query(selectTablesQuery() + whereClause, schemaRowsBuilder::withTables, true);
+    query(selectColumnsQuery() + whereClause, schemaRowsBuilder::withColumns, true);
     selectIndexesQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withIndexes));
+        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withIndexes, true));
     selectViewsQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withViews));
+        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withViews, true));
     selectFunctionsQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withFunctions));
+        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withFunctions, true));
     selectAggregatesQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withAggregates));
+        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withAggregates, true));
     selectVirtualKeyspacesQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withVirtualKeyspaces));
+        .ifPresent(
+            select -> query(select + whereClause, schemaRowsBuilder::withVirtualKeyspaces, false));
     selectVirtualTablesQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withVirtualTables));
+        .ifPresent(
+            select -> query(select + whereClause, schemaRowsBuilder::withVirtualTables, false));
     selectVirtualColumnsQuery()
-        .ifPresent(select -> query(select + whereClause, schemaRowsBuilder::withVirtualColumns));
+        .ifPresent(
+            select -> query(select + whereClause, schemaRowsBuilder::withVirtualColumns, false));
   }
 
   private void query(
       String queryString,
-      Function<Iterable<AdminRow>, CassandraSchemaRows.Builder> builderUpdater) {
+      Function<Iterable<AdminRow>, CassandraSchemaRows.Builder> builderUpdater,
+      boolean warnIfMissing) {
     assert adminExecutor.inEventLoop();
 
     pendingQueries += 1;
```

Comment:
the pending queries don't need to be an Atomic because everything is executed by the same thread, right?

Reply:
Yes, that's correct. All queries go through the same Netty I/O thread, and all other actions (sending the queries, processing the results) are also scheduled on that channel's event loop.